iam/x509-identity-mgmt/js/src/services/CertificateService.js

```diff
@@ -186,18 +191,42 @@ class CertificateService {
   async changeOwnership(filterFields, belongsTo = {}) {
     Object.assign(filterFields, { tenant: this.tenant });
 
-    // ensure that the current tenant owns the device
-    if (belongsTo.device) {
-      await this.ensureOwner(belongsTo.device);
-    }
+    // Checks on the certificate owner
+    await this.checkBelongsTo(belongsTo);
 
-    const result = await this.CertificateModel.findOneAndUpdate(
+    // By default, findOneAndUpdate() returns the document as it was before update was applied.
+    const certRecord = await this.CertificateModel.findOneAndUpdate(
       filterFields, { belongsTo, modifiedAt: new Date() },
-    ).maxTimeMS(this.queryMaxTimeMS).exec();
+    ).maxTimeMS(this.queryMaxTimeMS).lean().exec();
 
-    if (!result) {
+    if (!certRecord) {
       throw this.error.NotFound(`No records found for the following parameters: ${JSON.stringify(filterFields)}`);
     }
+
+    // -------------------------------------------------------------
+    // Checks what type of ownership notification needs to be issued
+    // -------------------------------------------------------------
+    const previousBelongsTo = certRecord.belongsTo;
+
+    if ((!previousBelongsTo.device && !previousBelongsTo.application)
+        && (belongsTo.device || belongsTo.application)) {
+      // If the certificate did not belong to any owner, but now
+      // it does, then we must notify the 'creation' of ownership
+      await this.ownershipNotifier.creation(certRecord, belongsTo);
+      // -----------------------------------------------------------
+    } else if ((previousBelongsTo.device || previousBelongsTo.application)
```

Comment:
Is there some previous validation to guarantee, for instance, that  `previousBelongsTo.device` is not equal to `belongsTo.device`?

Reply:
There isn't, but we can validate it! 
As the PATCH operation is idempotent, I believe that we can return successful execution (code 204), but not publish anything in kafka. What do you think?